module/dal/control/src/main/java/org/openbase/bco/dal/control/action/ActionImpl.java

```diff
@@ -728,6 +738,14 @@ private void cancelActionTask() {
         // cancel if exist
         if (actionTask != null) {
             actionTask.cancel(true);
+
+            // in case the task is directly cancel by this because the task was never canceled, we need to notify
+            // all waiting tasks about the cancellation
```

Comment:
```suggestion
            // if the task is canceled before it is scheduled, we need to notify
            // all waiting tasks about the cancellation
```

Good spot that we need to notify here as well!